### ISSUE TITLE
txmempool.cpp:498:85: warning: comparison between signed and unsigned integer expressions 

### ISSUE THREAD

**Issue by poiuty** [dashpay/dash#540]
```
  CXX    libbitcoin_server_a-txmempool.o
txmempool.cpp: In member function ‘void CTxMemPool::removeCoinbaseSpends(const CCoinsViewCache*, unsigned int)’:
txmempool.cpp:498:85: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

**Comment by jpentland** [comment]
This was fixed in bitcoin by Luke-Jr in this commit:

```
commit e617fe25785bfba30bcb0a8d8af6797b44ba07f5
Author: Luke Dashjr <luke-jr+git@utopios.org>
Date:   Wed May 13 20:04:39 2015 +0000

    Fix various warnings
    
    Found while building on Debian 7
```

```diff
-            if (!coins || (coins->IsCoinBase() && nMemPoolHeight - coins->nHeight < COINBASE_MATURITY)) {
+            if (!coins || (coins->IsCoinBase() && ((signed long)nMemPoolHeight) - coins->nHeight < COINBASE_MATURITY)) {
```
```

**Comment by jpentland** [comment]
Interestingly, the other two warnings fixed in this commit are applied in main.cpp and rest.h to variables which no longer exist. In the Linux kernel, single commits which fix multiple issues are not allowed, and this commit would've been broken up into three smaller commits to make them easier to apply in other forks and branches.

I will submit a pull request for this one line change only.

**Comment by schinzelh** [comment]
Many thanks

**Comment by schinzelh** [comment]
Merged to 0.12.1.x
